### PR TITLE
Fix tests for path validation changes on Unix

### LINKED ISSUE #25665: Many tests failing after path validation updates
Search for `[ActiveIssue(25665)]` in the tests.

### PR DESCRIPTION
**Pull request by Anipik**
Rebase on PR  #25688
Fixes #25665

### CODE REVIEW

File: src/System.IO.Compression.ZipFile/tests/ZipFileInvalidFileTests.cs [hunk 1]
```diff
@@ -213,14 +213,15 @@ public void DirectoryEntryWithData()
         /// This test ensures that a zipfile with path names that are invalid to this OS will throw errors
         /// when an attempt is made to extract them.
         /// </summary>
-        [ActiveIssue(25665)]
         [Theory]
         [InlineData("NullCharFileName_FromWindows")]
         [InlineData("NullCharFileName_FromUnix")]
         [PlatformSpecific(TestPlatforms.AnyUnix)]  // Checks Unix-specific invalid file path
-        public void Unix_ZipWithInvalidFileNames_ThrowsArgumentException(string zipName)
+        public void Unix_ZipWithInvalidFileNames(string zipName)
```
Comment (stephentoub) [inline src/System.IO.Compression.ZipFile/tests/ZipFileInvalidFileTests.cs]: But the names aren't invalid, are they?

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs [hunk 2]
```diff
@@ -33,16 +33,24 @@ public void EmptyAsPath_ThrowsArgumentException()
...
+                else if (invalidPath.Contains(@"\\?\") && !PathFeatures.IsUsingLegacyPathNormalization())
+                    Assert.Throws<DirectoryNotFoundException>(action);
+                else
+                    Assert.Throws<ArgumentException>(() => Create(invalidPath));
+            }
             else
-                Assert.Throws<ArgumentException>(() => Create(invalidPath));
+            {
+                // valid path for unix systems
+                //action();
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: We shouldn't have commented out code.  If this test shouldn't run on Unix, it should be annotated as such.

File: src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs [hunk 3]
```diff
@@ -33,16 +33,24 @@ public void EmptyAsPath_ThrowsArgumentException()
             Assert.Throws<ArgumentException>(() => Create(string.Empty));
         }
 
-        [ActiveIssue(25665)]
         [Theory, MemberData(nameof(PathsWithInvalidCharacters))]
         public void PathWithInvalidCharactersAsPath_ThrowsArgumentException(string invalidPath)
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs]: Seems like the test name is now wrong, too.  It's not necessarily throwing ArgumentException.

**Comment by stephentoub** [comment]
I started reviewing this, but stopped early on as it's difficult to review given that it includes all of the commits from the other pending PR.

**Comment by Anipik** [comment]
@dotnet-bot  test  Linux x64 Release Build

**Pull request closed without merging**